From b627c21978858501bbd2d39d08196e164139c159 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Tue, 5 Sep 2017 12:40:18 +0800
Subject: [PATCH 2503/5242] MLK-16370-1 video: fbdev: dcss: refine 'pitch'
 config logic for dpr

commit  f3351eff242a4f83a5c8b5c5cb842c9d943dff54 from
https://source.codeaurora.org/external/imx/linux-imx.git

Refine the 'pitch' configuration logic for DPR to
make it more reasonable and clear.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/imx_dcss.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/video/fbdev/mxc/imx_dcss.c b/drivers/video/fbdev/mxc/imx_dcss.c
index dc95646..017f3d2 100644
--- a/drivers/video/fbdev/mxc/imx_dcss.c
+++ b/drivers/video/fbdev/mxc/imx_dcss.c
@@ -1518,8 +1518,8 @@ static int dcss_dpr_config(uint32_t dpr_ch, struct dcss_info *info)
 
 	/* TODO: calculate pitch for different formats */
 	/* config pitch */
-	pitch = (var->xres * (var->bits_per_pixel >> 3)) << 16;
-	fill_sb(cb, chan_info->dpr_addr + 0x70, pitch);
+	pitch = var->xres * (var->bits_per_pixel >> 3);
+	fill_sb(cb, chan_info->dpr_addr + 0x70, pitch << 16);
 
 	fill_sb(cb, chan_info->dpr_addr + 0x200, 0x38);
 
-- 
1.7.9.5

